test/test_other.py

```diff
@@ -6430,12 +6430,25 @@ def test_modularize_sync_compilation(self):
     self.assertContained('''\
 before
 hello, world!
-object
-function
-undefined
+typeof result: object
+typeof _main: function
+typeof result.then: undefined
 after
 ''', self.run_js('a.out.js'))
 
+  def test_modularize_argument_misuse(self):
+    create_file('test.c', '''
+      #include <emscripten.h>
+      EMSCRIPTEN_KEEPALIVE int foo() { return 42; }''')
+
+    create_file('post.js', r'''
+      var arg = { bar: 1 };
+      var promise = Module(arg);
+      arg._foo();''')
+
+    expected = "Aborted(Access to module property ('_foo') is no longer possible via the incoming module contructor argument; Instead, use the result of the module promise"
```

Comment:
I read this comment a few times and I'm not actually sure what the user is expected to do to fix it? This is post-js code so it is inside the emitted JS, how can the module Promise be accessed?

Also, this seems like a breaking change that should be mentioned in the changelog?

Reply:
This is `--extern-post-js` code.. where we run the `Module()` constructor, but we ignore the return value and instead try to use a method on the argument we passed *into* the constructor.

Reply:
Ah, thanks, I missed the `-extern-`.